scripts/check-license-compliance.js

```diff
@@ -0,0 +1,103 @@
+#! /usr/bin/env node
+'use strict';
+/* eslint no-console: [ "error", { allow: [ "info", "warn", "error", "table" ] } ] */
+const { init } = require('license-checker'),
+      process = require('process'),
+      util = require('util'),
+      initLicenseChecker = util.promisify(init),
+      packageDirectory = require('pkg-dir');
+
+const ALLOWED_LICENSES = [
+   'MIT',
+   'ISC',
+   'Apache-2.0',
+   '(MIT OR Apache-2.0)',
+   'Public Domain',
+   'CC-BY-3.0',
+];
+
+(async () => {
+   const projectDirectoryPath = await packageDirectory(),
+         projectConfigPath = `${projectDirectoryPath}/.license-checker.cjs`,
+         licenseRecord = {};
+
+   let allowList = [],
+       moduleInfoReport,
+       unsupportedLicenses;
+
+   console.info('Project package.json located at path:', projectDirectoryPath);
+
+   try {
+      // eslint-disable-next-line global-require
+      const config = require(projectConfigPath);
+
+      allowList = config.allowList;
+
+      console.info(`Using configured allow-list at path ${projectConfigPath}`);
+      console.info('Allowing additional licenses:', allowList.join(', '));
+   } catch(e) {
+      console.error(e);
+      console.info('A config for license checker was not provided or was not found.');
+   }
+
+   try {
+      moduleInfoReport = await initLicenseChecker({
+         start: process.cwd(),
+      });
+   } catch(e) {
+      console.error('Could not initialize license-checker', e);
+      process.exitCode = 1;
+      return;
+   }
+
+   Object
+      .keys(moduleInfoReport)
+      .forEach((moduleName) => {
+         const licenses = moduleInfoReport[moduleName].licenses;
+
+         if (!licenses) {
+            return;
+         }
+
+         // normalize licenses since license-checker returns either a string or array
+         // of the license information
+         if (Array.isArray(licenses)) {
+            moduleInfoReport[moduleName].licenses.forEach((license) => {
+               licenseRecord[moduleName] = {
+                  module: moduleName,
+                  license: license,
+                  ...moduleInfoReport[moduleName],
+               };
+            });
```

Comment:
This only keeps the last license. Would it be better to have a list of all the licenses and see if any are in the list of permitted licenses?

Reply:
You're right! I neglected to revisit this block of code. Here's the situation: in the many projects I've tested it in, I've never seen `const licenses = moduleInfoReport[moduleName].licenses;` evaluate to an array.

That's probably because [the library is handling an edge case](https://github.com/davglass/license-checker/blob/master/lib/index.js#L124) due to [what the NPM docs say here](https://docs.npmjs.com/cli/v9/configuring-npm/package-json#license) regarding license data which they refer to as invalid.

In short, I don't think this block is actually needed from looking at the source linked above.

Reply:
I'm now simply taking whatever value it sends. Let me know if you think we should still support this case.